Sources/PackageRegistryTool/PackageRegistryTool+Publish.swift

```diff
@@ -0,0 +1,270 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Basics
+import Commands
+import CoreCommands
+import Foundation
+import PackageModel
+import PackageRegistry
+import TSCBasic
+import Workspace
+
+extension SwiftPackageRegistryTool {
+    struct Publish: SwiftCommand {
+        static let metadataFilename = "package-metadata.json"
+
+        static let configuration = CommandConfiguration(
+            abstract: "Publish to a registry"
+        )
+
+        @OptionGroup(visibility: .hidden)
+        var globalOptions: GlobalOptions
+
+        @Option(name: .customLong("id"), help: "The package identity")
+        var packageIdentity: PackageIdentity
+
+        @Option(help: "The package version")
+        var version: Version
+
+        @Option(name: .customLong("url"), help: "The registry URL")
+        var registryURL: URL?
+
+        @Option(
+            name: .customLong("output-directory"),
+            help: "The path of the directory where output file(s) will be written"
+        )
+        var customWorkingDirectory: AbsolutePath?
+
+        @Option(
+            name: .customLong("metadata-path"),
+            help: "The path to the package metadata JSON file"
+        )
+        var customMetadataPath: AbsolutePath?
+
+        @Option(help: "Signature format identifier. Defaults to 'cms-1.0.0'.")
```

Comment:
Should this option stay hidden for as long as it has this single signature format available? Otherwise it may lead to a confusion where users think it's configurable while it's not. Then when more cases become available in `SignatureFormat` we can unhide it.

```suggestion
        @Option(help: "Signature format identifier. Defaults to 'cms-1.0.0'.")
```

Reply:
@yim-lee wdyt? it is part of the spec

Reply:
> Should this option stay hidden for as long as it has this single signature format available?

Makes sense. I will update the proposal. Thanks for the suggestion.